Source/DafnyLanguageServer/Language/VerificationDafnyCodeActionProvider.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using Microsoft.Boogie;
+using Microsoft.Dafny.LanguageServer.Plugins;
+using Microsoft.Dafny.Plugins;
+using OmniSharp.Extensions.LanguageServer.Protocol.Models;
+using Range = OmniSharp.Extensions.LanguageServer.Protocol.Models.Range;
+
+namespace Microsoft.Dafny.LanguageServer.Language;
+
+/// <summary>
+/// A verification quick fixers provides quick "fixes" for verification errors.
+/// For now, it offers to inline a failing postcondition if there is no "return" keyword.
+/// </summary>
+class VerificationDafnyCodeActionProvider : DiagnosticDafnyCodeActionProvider {
+  protected override IEnumerable<DafnyCodeAction>? GetDafnyCodeActions(IDafnyCodeActionInput input, Diagnostic diagnostic, Range selection) {
+    var uri = input.Uri;
+    if (diagnostic.Source != MessageSource.Verifier.ToString()) {
+      return null;
+    }
+
+    if (diagnostic.RelatedInformation?.FirstOrDefault() is not { } relatedInformation) {
+      return null;
+    }
+
+    if (relatedInformation.Location.Uri != uri) {
+      return null;
+    }
+
+    var expression = DafnyCodeActionHelpers.Extract(relatedInformation.Location.Range, input.Code);
+    var (beforeEndBrace, indentationExtra, indentationUntilBrace) =
+      DafnyCodeActionHelpers.GetInformationToInsertAtEndOfBlock(input, diagnostic.Range.Start);
```

Comment:
I find the API of `DafnyCodeActionHelpers.GetInformationToInsertAtEndOfBlock` hard to understand. How about something like

```c#
static class RefactorTools {

  public DafnyCodeActionEdit? InsertBlockAtReturnLocation(
    DocumentTextBuffer text, 
    Range returnLocationRange,
    IEnumerable<string> statementsToInsert)

}
``` 
?

Reply:
Thanks for this refactoring suggestion, I applied it.
Note that I only need one statement to insert. For multiple statement, it's sufficient to call this function repeatedly and store the edits in the same order, because this is how they are going to be applied.